### PR TITLE
feat: extend the components model with has-like functions

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

As in description:
- Extend `Components` model with has-like functions:
  - hasMessages
  - hasSchemas
  - hasSecuritySchemes
  - hasParameters
  - hasCorrelationIds
  - hasOperationTraits
  - hasMessageTraits
- Extend `Schema` model with `property(name)` function to retrieve single property as Schema object -> idea from https://github.com/asyncapi/parser-js/issues/88#issuecomment-660054903
- Extend `AsyncAPIDocument` model with `hasDefaultContentType()` function
- Add relevant unit tests
- Add `test-lib` npm script for run only unit tests (without browser-test) - it's only for better dev-experience. Sometimes dev wants only write missing tests for models and do not want to run browser test every time -> in lib does not change anything itself.

**Related issue(s)**
Resolves https://github.com/asyncapi/parser-js/issues/173

### CODE REVIEW

**Comment by jonaslagoni** [comment]
@magicmatatjahu is this still in progress or?

**Comment by magicmatatjahu** [comment]
@jonaslagoni It's for the review :)

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
@magicmatatjahu major question I have is the use of `!!` in components and some testing comments 🙇 Are you sure the use of double question marks provide the expected return?

**Comment by magicmatatjahu** [comment]
@jonaslagoni Thanks for review. I understand your point of view about returned value -> should return false if array/object is empty or true, but please check that in other models in very similar methods (like in PR), if field of given model is an empty array or object, then function returns true, like here https://github.com/asyncapi/parser-js/blob/master/lib/models/server.js#L61 only in case undefined or null function returns false. We must discuss in which way we should go with @derberg @fmvilas. Please note that changing default behavior will be breaking change :) I will added suggested tests. Thanks again!

File: test/models/asyncapi_test.js [hunk 1]
```diff
@@ -58,6 +58,15 @@ describe('AsyncAPIDocument', function() {
...
 
+  describe('#serverNames()', function() {
+    it('should return an array of strings', function() {
+      const doc = { servers: { test1: { url: 'test1' }, test2: { url: 'test2' } } };
+      const d = new AsyncAPIDocument(doc);
+      expect(Array.isArray(d.serverNames())).to.be.equal(true);
+      expect(d.serverNames()).to.deep.equal(['test1', 'test2']);
+    });
+  });
+
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply test/models/asyncapi_test.js:69]: I will add this case in other functions, thanks! :)

File: lib/models/components.js [hunk 2]
```diff
@@ -54,8 +70,16 @@ class Components extends Base {
   securitySchemes() {
     return createMapOfType(this._json.securitySchemes, SecurityScheme);
   }
+
+  /**
+   * @returns {boolean}
+   */
+  hasSecuritySchemes() {
+    return !!this._json.securitySchemes;
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply lib/models/components.js:78]: Please check comment in main thread :)

**Comment by jonaslagoni** [comment]
> @jonaslagoni Thanks for review. I understand your point of view about returned value -> should return false if array/object is empty or true, but please check that in other models in very similar methods (like in PR), if field of given model is an empty array or object, then function returns true, like here https://github.com/asyncapi/parser-js/blob/master/lib/models/server.js#L61 only in case undefined or null function returns false. We must discuss in which way we should go with @derberg @fmvilas. Please note that changing default behavior will be breaking change :) I will added suggested tests. Thanks again!

Good point! What do you say we keep it as is for this PR and track it in another issue since it is, as you say, a major change?

**Comment by magicmatatjahu** [comment]
@jonaslagoni Sure, good idea :) I want that @derberg and @fmvilas know about our problem as well.

**Comment by derberg** [comment]
yeah, I think we already talked about this in the past, in some other issue or PR, so we definitely need a followup issue as we just need to discuss and plan how we change all over the place. As far as I remember I was fan of what @jonaslagoni wrote, about treating empty as false, but recently when working on template-for-generator-templates I liked a lot `has` functions and was missing them from some models a lot. Apporach with `has` has advantage over other solution, but yeah, something to discuss elsewhere

**Comment by jonaslagoni** [comment]
@magicmatatjahu will you or should I create the issue? Also just re-request review when you have added the corresponding tests :smile:

**Comment by magicmatatjahu** [comment]
@jonaslagoni I added missing tests only for checking returned empty object/array if given field doesn't exist in component or asyncapi model. Please check if everything is good :) If you want create issue, you're welcome 😄

File: test/models/asyncapi_test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply test/models/asyncapi_test.js]: Added.

File: test/models/asyncapi_test.js [hunk 3]
```diff
@@ -56,6 +56,29 @@ describe('AsyncAPIDocument', function() {
       expect(d.servers().test2.constructor.name).to.equal('Server');
       expect(d.servers().test2.json()).to.equal(doc.servers.test2);
     });
+
+    it('should return an empty object if the AsyncAPI document has not servers', function() {
```
Comment (jonaslagoni) [inline test/models/asyncapi_test.js:60]: Does this not make more sense as title for the test or is it the same? :thinking:  
```suggestion
    it('should return an empty object if the AsyncAPI document has no defined servers', function() {
```

**Review by jonaslagoni — APPROVED** [review]
Just a comment in regards to the titles for the tests. Not sure if its required to change that's up to you :smile: Good job  :+1:

File: test/models/asyncapi_test.js (hunk 3 above)
Comment (magicmatatjahu) [reply test/models/asyncapi_test.js]: Applied :)

**Review by jonaslagoni — APPROVED** [review]
:+1:

**Comment by derberg** [comment]
well done lads!
